tds-itemselection-aironline/src/main/java/tds/itemselection/algorithms/AdaptiveSelector2013.java

```diff
@@ -200,7 +209,32 @@ public ItemGroup selectNext (SQLConnection connection) throws ItemSelectionExcep
 	        }
 	        
 	        int n = Math.min(minitems, cset1.itemGroups.size());
-	        	
+
+			/*
+			 *   This section of code that occupies the following IF block was written to support Multi-Stage Braille
+			 *   assessments. The code in the MsbAssessmentSelectionService calls the adaptive algorithm and passes it a
+			 *   list of ItemGroup objects. Each of these item groups represents the entire contents of a fixed form
+			 *   segment (one segment per group). The groups replace the item pool remaining to the initial adaptive
+			 *   segment and force the algorithm to select a next "question" from the remaining item groups - in this
+			 *   case, the fixed form segment that best matches the student's current ability at the end of their
+			 *   adaptive section.
+			 */
+			if(itemGroups != null) {
+				CsetGroupCollection collection = new CsetGroupCollection();
+				ArrayList<CsetGroup> csetGroups = new ArrayList<>();
+				for(int i = 0; i < itemGroups.size(); i++) {
+					CsetGroup group = collection.setAndGet(itemGroups.get(i));
+					for(int j = 0; j < itemGroups.get(i).getItems().size(); j++) {
+						CSetItem item = new CSetItem(itemGroups.get(i).getItems().get(j), group);
+						group.addItem(item);
+					}
+					csetGroups.add(group);
+				}
+
+				cset1.itemGroups.clear();
+				cset1.itemGroups.addAll(csetGroups);
```

Comment:
Instead of clearing and adding can you just do `cset1.itemGroups = csetGroups;`?